pkg/providers/vsphere/setupuser/validator.go

```diff
@@ -0,0 +1,186 @@
+package setupuser
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/aws/eks-anywhere/pkg/validations"
+)
+
+const (
+	DefaultUsername   = "eksa"
+	DefaultGroup      = "EKSAUsers"
+	DefaultGlobalRole = "EKSAGlobalRole"
+	DefaultUserRole   = "EKSAUserRole"
+	DefaultAdminRole  = "EKSACloudAdminRole"
+)
+
+type connection struct {
+	Server   string `yaml:"server"`
+	Insecure bool   `yaml:"insecure"`
+}
+
+type objects struct {
+	Networks      []string `yaml:"networks"`
+	Datastores    []string `yaml:"datastores"`
+	ResourcePools []string `yaml:"resourcePools"`
+	Folders       []string `yaml:"folders"`
+	Templates     []string `yaml:"templates"`
+}
+
+type vSphereUserSpec struct {
+	Datacenter    string     `yaml:"datacenter"`
+	VSphereDomain string     `yaml:"vSphereDomain"`
+	Connection    connection `yaml:"connection"`
+	Objects       objects    `yaml:"objects"`
+	// Below are optional fields with defaults
+	Username   string `yaml:"username"`
+	GroupName  string `yaml:"group"`
+	GlobalRole string `yaml:"globalRole"`
+	UserRole   string `yaml:"userRole"`
+	AdminRole  string `yaml:"adminRole"`
+}
+
+type VSphereSetupUserConfig struct {
+	ApiVersion string          `yaml:"apiVersion"`
+	Kind       string          `yaml:"kind"`
+	Spec       vSphereUserSpec `yaml:"spec"`
+}
+
+func GenerateConfig(ctx context.Context, filepath string) (*VSphereSetupUserConfig, error) {
+	c, err := readConfig(ctx, filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = validate(ctx, c)
+	if err != nil {
+		return nil, err
+	}
+
+	setDefaults(ctx, c)
+
+	return c, nil
+}
+
+func readConfig(ctx context.Context, filepath string) (*VSphereSetupUserConfig, error) {
+	file, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read file %s, err = %v", filepath, err)
+	}
+
+	c := VSphereSetupUserConfig{}
+
+	if err = yaml.Unmarshal(file, &c); err != nil {
+		return nil, fmt.Errorf("Failed to parse %s, err = %v", filepath, err)
+	}
+
+	return &c, nil
+}
+
+func validate(ctx context.Context, c *VSphereSetupUserConfig) error {
+	results := []validations.ValidationResult{
```

Comment:
Use of the `ValidationResult` seems redundant here given it doesn't tie into some grander framework.

Reply:
Same answer here: https://github.com/aws/eks-anywhere/pull/3241#discussion_r962001331

Reply:
I think the 2 comments are a little differentiates. What I mean to say here is there's now consumer of `ValidationResult` in this package nor does anything return a set of validation results.

What we have here is semantically equivalent to:

```go
results := []error{
  validateDatacenter(...),
  validateDomain(...),
}

for _, result := range results {
  if result != nil {
    ...
  }
}
```